crypto/x509/x509_lcl.h

```diff
@@ -69,7 +69,7 @@ struct x509_crl_method_st {
 };
 
 struct x509_lookup_method_st {
-    const char *name;
+    char *name;
```

Comment:
So const is omitted because of it has to be eventually freed? Question is if it's more appropriate to cast it to non-const than to omit const here. I mean trade-off is between cast in free and masking all warnings that might indicate inappropriate/unintended use of this field... I for one leaning toward cast at free...

Reply:
I disagree, and this is more consistent with how we've done with other structures that were made opaque and for which we provided this kind of API.

Reply:
In my mind it's about contract. I mean if code is not *supposed* to modify this field through life-time of the object, then you can only win by telling compiler that it's not supposed to be modified. As for other occurrences, well, it's not the kind of thing that you can have one-size-fits-all... But maybe it would be appropriate to revisit even other cases...

Reply:
The trouble is that it isn't const.  You'd have to cast in `X509_LOOKUP_meth_new` as well, since the the given `name` is `strdup`'d.

Reply:
> You'd have to cast in X509_LOOKUP_meth_new as well, since the the given name is strdup'd.

No. Omission of const is problematic, assigning result of strdup to `const char *name` is not. Once again, const-ness is not necessarily about string residing in read-only segment, but how you treat it for its life time.

Reply:
I'm sorry, you're right.  But if we're going to disregard constness (by casting it away) when freeing, I'm not sure we should pretend there is constness...  This needs to be thought about, but I'd prefer to make that broad sweep among all the methods with titles that we provide.